libraries/plugins/elasticsearch/elasticsearch_plugin.cpp

```diff
@@ -488,4 +501,139 @@ void elasticsearch_plugin::plugin_startup()
    ilog("elasticsearch ACCOUNT HISTORY: plugin_startup() begin");
 }
 
+operation_history_object elasticsearch_plugin::get_operation_by_id(operation_history_id_type id)
+{
+   const string operation_id_string = std::string(object_id_type(id));
+
+   const string query = R"(
+   {
+      "query": {
+         "bool": {
+            "must": [
+            {
+               "query_string": {
+                  "query": "account_history.operation_id: )" + operation_id_string + R"("
+               }
+            },
+            {
+               "range": {
+                  "block_data.block_time": {
+                     "gte": "now-20y",
```

Comment:
Put a TODO in your calendar for the year 2034: bump this. ;-)

Can't you leave out this part of the query?

Reply:
I agree that the range could be omitted here.

Also I am wondering why you do query_string and not 
```
"query": {
        "match" : {
            "account_history.operation_id" : "operation_id_string"
        }
    }
```

Reply:
my apologies, this was not a production ready query but just a proof of concept probably pasted from kibana. i had changed this at https://github.com/bitshares/bitshares-core/pull/1725/commits/1f6ac147d5b298fecf17e7bc281676160192ea70

Reply:
for the second query i removed the range but query_string needs to stay as we are making a lucene query there and not just matching a field. https://github.com/bitshares/bitshares-core/pull/1725/commits/63f7affc81b0ca8cb274c97f2124d2d0d8d7ec05